contracts/SpokePool.sol

```diff
@@ -1339,12 +1339,7 @@ abstract contract SpokePool is
         // fillDeadline is relative to the destination chain.
         // Don’t allow fillDeadline to be more than several bundles into the future.
         // This limits the maximum required lookback for dataworker and relayer instances.
-        // Also, don't allow fillDeadline to be in the past. This poses a potential UX issue if the destination
-        // chain time keeping and this chain's time keeping are out of sync but is not really a practical hurdle
-        // unless they are significantly out of sync or the depositor is setting very short fill deadlines. This latter
-        // situation won't be a problem for honest users.
-        if (params.fillDeadline < currentTime || params.fillDeadline > currentTime + fillDeadlineBuffer)
-            revert InvalidFillDeadline();
+        if (params.fillDeadline > currentTime + fillDeadlineBuffer) revert InvalidFillDeadline();
```

Comment:
I just found one case where we assume fill deadline is not 0. Now, I don't think this is an issue but I'm gonna block this PR until we review more: https://github.com/across-protocol/sdk/blob/db244fbaec15dc049452cd104ca7bec5856ad3da/src/clients/BundleDataClient/BundleDataClient.ts#L817